internal/cmd/cmd_getdel.go

```diff
@@ -0,0 +1,92 @@
+package cmd
+
+import (
+	"github.com/dicedb/dice/internal/object"
+	dstore "github.com/dicedb/dice/internal/store"
+	"github.com/dicedb/dicedb-go/wire"
+)
+
+var cGETDEL = &DiceDBCommand{
+	Name:      "GETDEL",
+	HelpShort: "GETDEL returns the value for the queried key in args.",
+	Eval:      evalGETDEL,
+}
+
+func init() {
+	commandRegistry.AddCommand(cGETDEL)
+}
+
+// evalGETDEL returns the value for the queried key in args
+//
+// The function expects exactly one argument: the key to get.
+// If the key exists, it will be deleted before its value is returned.
+// evalGETDEL returns cmdResNil if key is expired or it does not exist
+//
+// Parameters:
+//   - c *Cmd: The command context containing the arguments
+//   - s *dstore.Store: The data store instance
+//
+// Returns:
+//   - *CmdRes: Response containing the value for the queried key.
+//   - error: Error if wrong number of arguments or wrong value type.
+func evalGETDEL(c *Cmd, s *dstore.Store) (*CmdRes, error) {
+	if len(c.C.Args) != 1 {
+		return cmdResNil, errWrongArgumentCount("GETDEL")
+	}
+
+	key := c.C.Args[0]
+
+	// Getting the key based on previous touch value
+	obj := s.GetNoTouch(key)
+
+	// If key does not exist, return nil
+	if obj == nil {
+		return cmdResNil, nil
+	}
+
+	// If the object exists, check if it is either a JSON or SET object.
+	switch obj.Type {
+	case object.ObjTypeSet, object.ObjTypeJSON:
+		return cmdResNil, errWrongTypeOperation("GETDEL")
+	}
```

Comment:
why this check? we can safely remove them. DiceDB will return the value as is and then will delete the key.